thrift/src/main/java/com/linecorp/armeria/common/thrift/text/StackWalkingThriftMessageClassFinder.java

```diff
@@ -0,0 +1,139 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+// =================================================================================================
+// Copyright 2011 Twitter, Inc.
+// -------------------------------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this work except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file, or at:
+//
+//  https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =================================================================================================
+package com.linecorp.armeria.common.thrift.text;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.invoke.MethodType;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class StackWalkingThriftMessageClassFinder extends AbstractThriftMessageClassFinder {
+    private final Logger logger =
+            LoggerFactory.getLogger(StackWalkingThriftMessageClassFinder.class);
+
+    @Nullable
+    private final Function<Stream<Object>, Class<?>> walkHandler;
+    private final MethodHandle walkMH;
+    private final Object stackWalker;
+
+    StackWalkingThriftMessageClassFinder() throws Throwable {
+        final ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        final Lookup lookup = MethodHandles.lookup();
+
+        final Class<?> stackWalkerClass = classLoader.loadClass("java.lang.StackWalker");
+        walkMH = lookup.findVirtual(stackWalkerClass,
+                                    "walk",
+                                    MethodType.methodType(Object.class, Function.class));
+
+        final Class<?> stackFrameClass = classLoader.loadClass("java.lang.StackWalker$StackFrame");
+        final String invokingFailMsg = "Failed to invoke StackWalker.StackFrame.getDeclaringClass():";
```

Comment:
I think this is useless if an exception is not thrown. How about embedding into loggger.warn(...) with duplication?

```
logger.warn("Failed to invoke StackWalker.StackFrame.getDeclaringClass():", t);
```